repository/managers.py

```diff
@@ -1,51 +1,99 @@
+
+    def get_item_details(self, group_id, item_id, params=None):
+        """Fetch detailed information of an item with optional parameters."""
+        return self._make_request(f"/groups/{group_id}/items/{item_id}/", params=params)
+
+
+class RepositoryManager:
+    def __init__(self, user, repository):
+        """Initialize the manager with the user and repository."""
+        self.api = CitesphereAPIv1(user, repository)
+        self.user = user
+        self.repository = repository
 
     def get_raw(self, target, **params):
-        headers = {}
-        if self.user and self.repository:
-            headers = auth.citesphere_auth(self.user, self.repository)
-        return requests.get(target, headers=headers, params=params).content
+        """Fetch raw data from any API target."""
+        try:
+            response = requests.get(target, headers=self.api._get_headers(), params=params)
+            response.raise_for_status()
+            return response.content
+        except RequestException as e:
+            raise CitesphereAPIError(message="Failed to fetch data", error_code="RAW_DATA_ERROR", details=str(e))
 
     def groups(self):
-        """Fetch Groups from the repository's endpoint"""
-        headers = auth.citesphere_auth(self.user, self.repository)
-        url = f"{self.repository.endpoint}/api/v1/groups/"
-        response = requests.get(url, headers=headers)
-        
-        if response.status_code == 200:
-            return response.json()  # Return the groups data
-        else:
-            response.raise_for_status()
+        """Fetch all groups from the repository."""
+        return self.api.get_groups()
 
-    def group_items(self, groupId, page=1):
+    def group_items(self, group_id, page=1):
         """
         Fetch items from a specific group for a specific page.
-        """
-        headers = auth.citesphere_auth(self.user, self.repository)
-        base_url = f"{self.repository.endpoint}/api/v1/groups/{groupId}/items/"
 
-        params = {
-            'page': page,
-        }
+        Args:
+            group_id: The ID of the group in the repository.
+            page: The page number to retrieve.
 
-        group_response = requests.get(base_url, headers=headers, params=params)
-        group_response.raise_for_status()
+        Returns:
+            A dictionary containing:
+                - "group": Details about the group.
+                - "items": A list of items in the group for the specified page.
+                - "total_items": The total number of items in the group.
+                
+        Raises:
+            CitesphereAPIError
+        """
+        if not isinstance(page, int) or page < 1:
+            raise CitesphereAPIError(message="Invalid page number", error_code="INVALID_PAGE", details="Page must be a positive integer")
 
-        response_data = group_response.json()
+        response_data = self.api._make_request(f"/groups/{group_id}/items/", params={'page': page})
```

Comment:
this should be part of the citesphere class (the path), since it's specific to citesphere.